compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/KernelDispatch.cpp

```diff
@@ -422,74 +498,46 @@ static LogicalResult setRootConfigForPadPackPipeline(
   return success();
 }
 
-/// TODO(avarma): This currently is skipping checking for ext* ops.
-static bool bodyMatcherForMatmulTranspose(Value yieldVal, Block *body) {
-  Operation *addOp = yieldVal.getDefiningOp();
-  if (!isa_and_nonnull<arith::AddIOp, arith::AddFOp>(addOp)) {
-    return false;
-  }
-  Operation *mulOp = addOp->getOperand(1).getDefiningOp();
-  if (!isa_and_nonnull<arith::MulIOp, arith::MulFOp>(mulOp)) {
-    return false;
-  }
-  auto lhsBlockArg = dyn_cast<BlockArgument>(mulOp->getOperand(0));
-  auto rhsBlockArg = dyn_cast<BlockArgument>(mulOp->getOperand(1));
-  auto outBlockArg = dyn_cast<BlockArgument>(addOp->getOperand(0));
-  if (!lhsBlockArg || !rhsBlockArg || !outBlockArg ||
-      lhsBlockArg.getOwner() != body || rhsBlockArg.getOwner() != body ||
-      outBlockArg.getOwner() != body || lhsBlockArg.getArgNumber() != 0 ||
-      rhsBlockArg.getArgNumber() != 1 || outBlockArg.getArgNumber() != 2) {
-    return false;
-  }
-  return true;
-}
-
-/// `isMatmulTransposeB` is a utility function that aims to indentify whether a
-/// linalg.generic op is a matmul with rhs operand transposed.
-static bool isMatmulTransposeB(linalg::GenericOp genericOp) {
-  // Step 1. Test the body of the generic to indeed be what we expect for a
-  //         matmul transpose.
-  Block *body = genericOp.getBlock();
-  auto yieldOp = cast<linalg::YieldOp>(body->getTerminator());
-  Value yieldVal = yieldOp.getOperand(0);
-  if (!bodyMatcherForMatmulTranspose(yieldVal, body)) {
-    return false;
-  }
-  // Step 2. Check iterator types.
-  SmallVector<utils::IteratorType> matmulTransposeIteratorTypes = {
-      utils::IteratorType::parallel, utils::IteratorType::parallel,
-      utils::IteratorType::reduction};
-  SmallVector<utils::IteratorType> opIteratorTypes =
-      genericOp.getIteratorTypesArray();
-  if (matmulTransposeIteratorTypes != opIteratorTypes) {
-    return false;
+static LogicalResult setRootConfigForConvDecomposePipeline(
+    mlir::FunctionOpInterface entryPointFn, linalg::LinalgOp linalgOp,
+    AIEConfig cfg) {
+  auto maybeInstructionSize = getMatmulInstructionSize(linalgOp);
```

Comment:
Expand `auto` here and elsewhere.

Here: `auto` -> `FailureOr<std::array<uint32_t, 3>>`

Reply:
I prefer auto ;) Do we have a coding style reference?

Reply:
The compiler follows llvm standards so this
https://releases.llvm.org/10.0.0/docs/CodingStandards.html#use-auto-type-deduction-to-make-code-more-readable

It is subjective on if it makes the code more readable in this case :) 
Since the llvm standard leaves room for interpretation we can also look athe google standard (which the non compiler part of the project follow)
https://google.github.io/styleguide/cppguide.html#Type_deduction

Reply:
I think this is a good guideline. Thanks for sharing it. 
```Don’t “almost always” use auto, but do use auto with initializers like cast<Foo>(...) or other places where the type is already obvious from the context. Another time when auto works well for these purposes is when the type would have been abstracted away anyways, often behind a container’s typedef such as std::vector<T>::iterator.```